### PR TITLE
add replica resize operation to ReplicaOperations 

### PR DESCRIPTION
**Pull request by dsharma-dc**
This will be mainly used for test purpose, and not in the normal prod workflow.

### CODE REVIEW

**Comment by dsharma-dc** [comment]
bors try

**Comment by dsharma-dc** [comment]
bors try

File: control-plane/grpc/src/operations/replica/traits.rs [hunk 1]
```diff
@@ -54,6 +53,12 @@ pub trait ReplicaOperations: Send + Sync {
         req: &dyn UnshareReplicaInfo,
         ctx: Option<Context>,
     ) -> Result<(), ReplyError>;
+    /// Resize a replica
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/replica/traits.rs]:
```suggestion
    /// Resize a replica.
```

File: control-plane/grpc/proto/v1/replica/replica.proto [hunk 2]
```diff
@@ -157,6 +157,19 @@ message UnshareReplicaRequest {
   optional google.protobuf.StringValue pool_uuid = 5;
 }
 
+// Resize a replica to the requested size
+// For internal usage mostly.
+message ResizeReplicaRequest {
+  // id of the io-engine instance
+  string node_id = 1;
+  // uuid of the replica
+  google.protobuf.StringValue replica_id = 2;
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/replica/replica.proto]:
```suggestion
  string replica_id = 2;
```

File: control-plane/grpc/proto/v1/replica/replica.proto [hunk 3]
```diff
@@ -157,6 +157,19 @@ message UnshareReplicaRequest {
...
+  // requested new size of the volume
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/replica/replica.proto]:
```suggestion
  // requested new size for the replica
```

File: control-plane/grpc/proto/v1/replica/replica.proto [hunk 4]
```diff
@@ -157,6 +157,19 @@ message UnshareReplicaRequest {
...
+  uint64 requested_size = 3;
+    // id of the pool
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/replica/replica.proto]:
```suggestion
  // id of the pool
  string pool_id = 4;
  // uuid of the pool
  optional string pool_uuid = 5;
```

**Review by tiagolobocastro — COMMENTED** [review]
What is the intended use case for this replica size?
Do we want to allow this for replicas which are part of a volume? What is the use case?

**Comment by dsharma-dc** [comment]
> What is the intended use case for this replica size? Do we want to allow this for replicas which are part of a volume? What is the use case?

This is primarily added so we could test reconciler. It's not expected to be used for normal volume resize workflows.

File: control-plane/grpc/proto/v1/replica/replica.proto (hunk 2 above)
Comment (dsharma-dc) [reply control-plane/grpc/proto/v1/replica/replica.proto]: @tiagolobocastro All the messages in this file have replica_id as StringValue. Any particular reason to make that `string` here?

File: control-plane/grpc/proto/v1/replica/replica.proto (hunk 4 above)
Comment (dsharma-dc) [reply control-plane/grpc/proto/v1/replica/replica.proto]: There is no `pool_uuid` in the transport's `ResizeReplica` type as of today, hence didn't keep it here.

Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/replica/replica.proto]: interesting, there should be one there..

Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/replica/replica.proto]: ah I see we already have a mix of having uuid and not having it :( 
So seems like we can skip the pool altogether here as per existing requests

File: control-plane/grpc/proto/v1/replica/replica.proto (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/replica/replica.proto]: they were added mistakenly... we've since switched to string

Comment (dsharma-dc) [reply control-plane/grpc/proto/v1/replica/replica.proto]: ok, will change. I guess having it `string` will help simplify code as we may not need `Validated*` type in between.

File: control-plane/grpc/src/operations/replica/traits.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/grpc/src/operations/replica/traits.rs]: Did you forget to push? This is resolved but changes are not visible

Comment (dsharma-dc) [reply control-plane/grpc/src/operations/replica/traits.rs]: > Did you forget to push? This is resolved but changes are not visible

Yeah it wasn't pushed yet. Done now.

File: control-plane/grpc/src/operations/replica/traits.rs [hunk 5]
```diff
@@ -782,6 +788,86 @@ impl ValidateRequestTypes for UnshareReplicaRequest {
     }
 }
 
+/// ResizeReplicaInfo trait for the replica resize to be implemented by entities which want to
+/// avail this operation
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/replica/traits.rs]: please terminate all doc comments with `.`

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/src/operations/replica/traits.rs (hunk 5 above)
Comment (dsharma-dc) [reply control-plane/grpc/src/operations/replica/traits.rs]: Done

**Review by hrudaya21 — APPROVED** [review]

**Comment by dsharma-dc** [comment]
bors merge

**Pull request merged**